consensus/istanbul/backend/announce.go

```diff
@@ -209,29 +274,47 @@ func (sb *Backend) handleIstAnnounce(payload []byte) error {
 	// Even if it is deployed, it may not have any registered validators yet.
 	if err == errValidatorsContractNotRegistered || len(regVals) == 0 {
 		sb.logger.Trace("Can't retrieve the registered validators.  Only allowing the initial validator set to send announce messages", "err", err, "regVals", regVals)
-		block := sb.currentBlock()
-		valSet := sb.getValidators(block.Number().Uint64(), block.Hash())
-
 		regVals = make(map[common.Address]bool)
-		for _, val := range valSet.List() {
-			regVals[val.Address()] = true
-		}
 	} else if err != nil {
 		sb.logger.Error("Error in retrieving the registered validators", "err", err)
 		return err
 	}
 
+	// Add active validators regardless
+	block := sb.currentBlock()
+	valSet := sb.getValidators(block.Number().Uint64(), block.Hash())
+	for _, val := range valSet.List() {
+		regVals[val.Address()] = true
+	}
+
 	if !regVals[msg.Address] {
-		sb.logger.Warn("Received an IstanbulAnnounce message from a non registered validator. Ignoring it.", "AnnounceMsg", msg.String())
+		sb.logger.Warn("Received an IstanbulAnnounce message from a non registered validator. Ignoring it.", "AnnounceMsg", msg.String(), "validators", regVals, "err", err)
 		return errUnauthorizedAnnounceMessage
 	}
 
+	// Decrypt the EnodeURL
+	nodeKey := ecies.ImportECDSA(sb.GetNodeKey())
+
+	encryptedIP := []byte("")
+	for _, entry := range msg.EncryptedIPData {
+		if bytes.Equal(entry[0], sb.Address().Bytes()) {
+			encryptedIP = entry[1]
+		}
+	}
+	IPBytes, err := nodeKey.Decrypt(encryptedIP, nil, nil)
+	if err != nil {
+		sb.logger.Warn("Error in decrypting ip", "err", err)
+	}
+	enodeUrl := msg.IncompleteEnodeURL + string(IPBytes)
+
+	log.Error("Handling an IstanbulAnnounce message", "actual", msg.IncompleteEnodeURL, "lol", enodeUrl)
```

Comment:
nit:  LOL!  Also, please change severity to trace.

Reply:
also forgot to remove!